aiida/repository/backend/abstract.py

```diff
@@ -105,6 +106,21 @@ def get_object_content(self, key: str) -> bytes:
         with self.open(key) as handle:
             return handle.read()
 
+    def get_object_hash(self, key: str) -> str:
+        """Return the hash of a object identified by key.
+
+        .. note::
+            This hash value is dependant on the repository implementation
+            and so may change for different repositories.
+            By default the algorithm is SHA-256
+
+        :param key: fully qualified identifier for the object within the repository.
+        :raise FileNotFoundError: if the file does not exist.
+        :raise OSError: if the file could not be opened.
+        """
+        with self.open(key) as handle:
+            return hashlib.sha256(handle.read()).hexdigest()
```

Comment:
This might exhaust RAM if the object is very big (possible).

You should compute the hash by reading the file in chunks.
You can copy-paste and adapt from here as an example:
https://github.com/aiidateam/disk-objectstore/blob/a2561a40cf2e9a4d58325387f913ccf08111f5fd/disk_objectstore/utils.py#L815-L839

(It's possible we've the same issue elsewhere in AiiDA, but I would start improving those one by one)

Reply:
cheers will do

Reply:
Implemented `aiida.common.hashing.chunked_file_hash`